osu.Game/Rulesets/UI/HitObjectContainer.cs

```diff
@@ -15,5 +16,15 @@ public class HitObjectContainer : CompositeDrawable
 
         public virtual void Add(DrawableHitObject hitObject) => AddInternal(hitObject);
         public virtual bool Remove(DrawableHitObject hitObject) => RemoveInternal(hitObject);
+
+        protected override int Compare(Drawable x, Drawable y)

```

Comment:
This doesn't affect depth any more, does it?

Reply:
It does - later hitobjects will be below earlier hitobjects.